resolver/resolver.cc

```diff
@@ -1493,14 +1493,18 @@ class ResolveSignaturesWalk {
             return asgn;
         }
 
-        if (data->isStaticField() && data->resultType == nullptr) {
-            data->resultType = resolveConstantType(ctx, asgn->rhs, sym);
+        if (data->isStaticField()) {
             if (data->resultType == nullptr) {
-                auto rhs = move(asgn->rhs);
-                auto loc = rhs->loc;
-                asgn->rhs = ast::MK::Send1(loc, ast::MK::Constant(loc, core::Symbols::Magic()),
-                                           core::Names::suggestType(), move(rhs));
-                data->resultType = core::Types::untyped(ctx, sym);
+                data->resultType = resolveConstantType(ctx, asgn->rhs, sym);
+                if (data->resultType == nullptr) {
+                    auto rhs = move(asgn->rhs);
+                    auto loc = rhs->loc;
+                    asgn->rhs = ast::MK::Send1(loc, ast::MK::Constant(loc, core::Symbols::Magic()),
+                                               core::Names::suggestType(), move(rhs));
+                    data->resultType = core::Types::untyped(ctx, sym);
+                }
+            } else if (!core::isa_type<core::AliasType>(&*data->resultType)) {
```

Comment:
```suggestion
            } else if (!core::isa_type<core::AliasType>(data->resultType.get())) {
```

Can we leave a comment explaining this branch? It's not intuitive for me